### PR TITLE
Remove SLI examples in model documentation

### PR DESCRIPTION
**Pull request by jessica-mitchell**
I stumbled upon a couple of SLI examples that are still lingering in the model documentation - in spike_generator and music_rate_out_proxy specifically.
This PR simply removes them, and fixes up a bit of formatting.

### CODE REVIEW

**Review by heplesser — CHANGES_REQUESTED** [review]
@jessica-mitchell The rules for moving spikes times to grid time points in `spike_detector` are not trivial. The examples given in the code provide valuable information which should not get lost. The principles are described in the remaining documentation, but they become a lot easier to understand with the examples in place. How about converting the SLI examples to corresponding PyNEST examples?

**Comment by jessica-mitchell** [comment]
@heplesser I converted the SLI examples to PyNEST, please take a look that I did not bugger the syntax and it makes sense in the context of PyNEST.

File: models/spike_generator.h [hunk 1]
```diff
@@ -79,72 +79,101 @@ combination of step and offset. This should only be used if all neurons
...
+return different `spike_times` values at different resolutions.
 
 Example:
 
-    spike_generator << /spike_times [1.0 2.0 3.0] >> SetStatus
+  ::
 
-    Instructs the spike generator to generate events at 1.0, 2.0, and
-    3.0 milliseconds, relative to the device-timer origin.
+     nest.SetStatus("spike_generator",
```
Comment (heplesser) [inline models/spike_generator.h:106]:
```suggestion
     nest.Create("spike_generator",
```

**Review by heplesser — CHANGES_REQUESTED** [review]
@jessica-mitchell Thanks for the PyNEST example. There is one little error for which I added a suggestion. One general question: Have we made a decision concerning the use of single vs double quotes in Python code?

**Comment by jessica-mitchell** [comment]
@heplesser I dont remember for certain how discussions ended- but i think we were leaning towards double quotes.

**Review by heplesser — APPROVED** [review]

File: models/spike_generator.h [hunk 2]
```diff
@@ -79,72 +79,101 @@ combination of step and offset. This should only be used if all neurons
...
+               "precise_times": True})
+
+  ---> spikes at step 10, offset 0.0; step 11, offset -0.05;
          step 31, offset -0.0999
 
-    Assume we have simulated 10.0ms and simulation times is thus 10.0 (step
-100).
-    Then, any spike times set, at this time, must be later than step 100.
+Assume we have simulated 10.0 ms and simulation times is thus 10.0 (step
+100). Then, any spike times set, at this time, must be later than step 100.
```
Comment (sarakonradi) [inline models/spike_generator.h:151]:
```suggestion
100). Then, any spike times set at this time must be later than step 100.
```

File: models/spike_generator.h [hunk 3]
```diff
@@ -79,72 +79,101 @@ combination of step and offset. This should only be used if all neurons
...
+               params={"spike_times": [1.0, 1.05, 3.0001],
+               "precise_times": True})
+
+  ---> spikes at step 10, offset 0.0; step 11, offset -0.05;
          step 31, offset -0.0999
 
-    Assume we have simulated 10.0ms and simulation times is thus 10.0 (step
-100).
-    Then, any spike times set, at this time, must be later than step 100.
+Assume we have simulated 10.0 ms and simulation times is thus 10.0 (step
```
Comment (sarakonradi) [inline models/spike_generator.h:150]: Do you mean ``simulation time``?

File: models/spike_generator.h [hunk 4]
```diff
@@ -79,72 +79,101 @@ combination of step and offset. This should only be used if all neurons
...
+                 params={"spike_times": [1.0, 2.0, 3.0]})
 
-Example:
+  Instructs the spike generator to generate events at 1.0, 2.0, and
+  3.0 milliseconds, relative to the device-timer origin.
 
-Assume that NEST works with default resolution (step size) of 0.1ms
-and default tic length of 0.001ms. Then, spikes times not falling
+Example:
+Assume that NEST works with default resolution (step size) of 0.1 ms
```
Comment (sarakonradi) [inline models/spike_generator.h:113]:
```suggestion
Example:

Assume that NEST works with default resolution (step size) of 0.1 ms
```

**Review by sarakonradi — CHANGES_REQUESTED** [review]
@jessica-mitchell Thanks! I have added a few minor suggestions inline.

File: models/spike_generator.h (hunk 3 above)
Comment (jessica-mitchell) [reply models/spike_generator.h:151]:
```suggestion
Assume we have simulated 10.0 ms and simulation time is thus 10.0 (step
```

**Comment by jessica-mitchell** [comment]
@sarakonradi I've accepted all your suggestions

**Review by sarakonradi — APPROVED** [review]

**Comment by heplesser** [comment]
This PR changes only comments; the failing test is related to #1944 and definitely not related to this PR. I will this merge now.

**Pull request merged**